streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamImplTest.java

```diff
@@ -651,13 +652,15 @@ public void shouldThrowNullPointerOnJoinWithTableWhenJoinedIsNull() {
     }
 
     @Test(expected = NullPointerException.class)
-    public void shouldThrowNullPointerOnJoinWithStreamWhenJoinedIsNull() {
-        testStream.join(testStream, MockValueJoiner.TOSTRING_JOINER, JoinWindows.of(ofMillis(10)), null);
+    public void shouldThrowNullPointerOnJoinWithStreamWhenStreamJoinedIsNull() {
+        final StreamJoined joined = null;
+        testStream.join(testStream, MockValueJoiner.TOSTRING_JOINER, JoinWindows.of(ofMillis(10)), joined);
```

Comment:
super-nit: did this variable (`joined`) get added during auto-refactoring? It seems... unnecessary.

Reply:
Needed to disambiguate the test from `Joined` or `StreamJoined`, cleaned up some to use an instance variable.

Reply:
Ah, that makes sense... I've had that happen a bunch in tests, we can just cast the null: `(StreamJoined) null`. But the variable is fine, too. Just threw me a little.